### PR TITLE
Add Unit Test for Web3JsonRPC Actions

### PR DESCRIPTION
**Pull request by AMITGOELNY**
### _Summary_
Add Unit Test for Web3JsonRPC Actions

### _How did you test your changes?_
Ran tests locally

### CODE REVIEW

File: android/walletsdk/src/main/java/com/coinbase/android/nativesdk/message/request/Action.kt [hunk 1]
```diff
@@ -20,4 +20,21 @@ class Action {
...
+                    this.optional == other.optional
+        }
+    }
+
+    override fun hashCode(): Int {
+        var result = method.hashCode()
+        result = 31 * result + paramsJson.hashCode()
+        result = 31 * result + optional.hashCode()
+        return result
+    }
```
Comment (vishnumad) [inline android/walletsdk/src/main/java/com/coinbase/android/nativesdk/message/request/Action.kt:39]: Worth changing Actions to be data classes?

File: android/walletsdk/src/test/java/com/coinbase/android/nativesdk/message/request/Web3JsonRPCTest.kt [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    fun action_Test_Personal_Sign_Action() {
+        val inputStream = javaClass.classLoader?.getResourceAsStream("personal_sign.json")
+        val personalSignJson = inputStream.readFileWithNewLineFromResources()
+        val expectedAction = Action(
+            method = "personal_sign",
+            paramsJson = personalSignJson,
+            optional = false
+        )
+
+        val action = Web3JsonRPC.PersonalSign("0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826", "hello").action()
```
Comment (vishnumad) [inline android/walletsdk/src/test/java/com/coinbase/android/nativesdk/message/request/Web3JsonRPCTest.kt:34]: Personal sign message here should be a hex string I believe: `hello` -> `0x68656c6c6f`

**Review by vishnumad — APPROVED** [review]

File: android/walletsdk/src/main/java/com/coinbase/android/nativesdk/message/request/Action.kt (hunk 1 above)
Comment (AMITGOELNY) [reply android/walletsdk/src/main/java/com/coinbase/android/nativesdk/message/request/Action.kt:39]: Not sure what the impact will be downstream, so avoided that.  If you think it's safe, I guess we can make that change

File: android/walletsdk/src/test/java/com/coinbase/android/nativesdk/message/request/Web3JsonRPCTest.kt (hunk 2 above)
Comment (AMITGOELNY) [reply android/walletsdk/src/test/java/com/coinbase/android/nativesdk/message/request/Web3JsonRPCTest.kt:34]: We're using the raw string for message in personal sign, even in our docs.

**Pull request merged**